src/coreclr/utilcode/executableallocator.cpp

```diff
@@ -66,7 +66,12 @@ void ExecutableAllocator::InitLazyPreferredRange(size_t base, size_t size, int r
     // to coreclr.dll.  This avoids having to create jump stubs for calls to
     // helpers and R2R images loaded close to coreclr.dll.
     //
+#ifdef TARGET_XARCH
     SIZE_T reach = 0x7FFF0000u;
+#else
+    // Smaller size for ARM64 where relative calls/jumps only work within 128MB
+    SIZE_T reach = 0x7FF0000u;
```

Comment:
This may collide with other ASLR assigned addresses and lead to non-trivial private working set hits for native .dlls or R2R images that are loaded after coreclr is initialized.

We try to start as far as possible from coreclr base address to avoid that situation on x64. Look for the comment below: "We try to occupy the space as far as possible to minimize collisions with other ASLR assigned addresses."

Reply:
Are you saying that win-arm64 is fine as is?

I am still trying to understand the problem. E.g. we reserve 2GB via, I assume, VirtualAlloc
how come we end up with a large distance between `coreclr` code and jitted code in memory?

Reply:
Just checked - win-arm64 app emits less jump-stubs for a completely empty app: only 4 (two inside StelemRef and two inside IndexOf)

Reply:
> Are you saying that win-arm64 is fine as is?

I am saying that this change is potentially replacing one performance problem with a different performance problem on win-arm64.

All modern OSes have address space layout randomization. Our attempts to allocate near coreclr library are going to against that. So we have to be careful not to be on collision course with what the OSes are trying to do.

> win-arm64 app emits less jump-stubs for a completely empty app

Before this change, or only with this change?

Reply:
> Before this change, or only with this change?

Before (Main)

Reply:
Do you understand why it is the case? The executable space should be typically allocated more than 128MB away from coreclr.dll on win-arm64 if I am reading this code correctly.

Reply:
@jkotas oops, nvm, I forgot that on M1 I used `DOTNET_ReadyToRun=0`. I've just tried it with R2R on win-arm64 and got exactly the same list of methods requesting jump stubs as in https://github.com/dotnet/runtime/issues/62302#issuecomment-1013855517